### ISSUE TITLE
[Feature] New test status for screenshot tests (something like passedWithDiff)

### ISSUE THREAD

**Issue by bezyakina** [microsoft/playwright#14485]
Hi! We have screenshot tests that may have a small pixel diff or they may fall before reaching the screenshot. We would like to be able to repeat the test if it failed not because of the difference in pixels. And also we don't want to retry the test if the error is caused due to pixel difference.

I see 2 options here:
1. Add a new status for screenshot tests (something like passedWithDiff)
2. Add the ability to enable/disable the retries of tests directly from the test, suggested here https://github.com/microsoft/playwright/issues/11584

Thank you for your attention)

**Comment by aslushnikov** [comment]
@bezyakina why don't you want to retry tests in case of screenshot differences? Is that to save some time?

**Comment by bezyakina** [comment]
@aslushnikov
Because we take long screenshots, out of the box full-screen screenshots don't work correctly with mobile webviews, so we have to scroll, then take a screenshot, repeat and at the end combine all the screenshots into one big screenshot. This operation can take up to a minute. Since the tests are stable, if the layout is broken, then there is no point in repeating the test, because we will get the same result.

**Comment by aslushnikov** [comment]
> Since the tests are stable, if the layout is broken, then there is no point in repeating the test, because we will get the same result.

@bezyakina I see. So far test retries cannot be configured dynamically, and there are no immediate plans to do so.

The best workaround I can think of is to split out the screenshot tests into a separate project and configure it with no-retries policy. Would it work?

**Comment by bezyakina** [comment]
@aslushnikov
We did it. There are only screenshot tests in our suite. But when the tests fall due to unstable devices connection, we will have to restart the entire test suite. The point was not to restart the entire suite. Instead, retry the only failed test directly.

**Comment by pavelfeldman** [comment]
I would probably try/catch the screenshot comparison. It will still produce the artifacts, but will not fail the execution. We can consider adding `expect.trial` to mitigate your use case more broadly.

**Comment by dgozman** [comment]
@bezyakina Did you try the [try/catch workaround](https://github.com/microsoft/playwright/issues/14485#issuecomment-1147941557)? If so, how was your experience? I am evaluating whether we should to `expect.trial` or not, and if we do whether that would actually help you.

**Comment by bezyakina** [comment]
@dgozman  Yes, I tried try/catch, if the error message contained the word like "diff", I changed the test status to passed. But in this way, it is difficult to separate the passed tests from those passed with a small diff in an html report.

**Comment by mscottford** [comment]
I created #23197 which might be a duplicate of this issue.

**Comment by joergRossdeutscher** [comment]
Totally agree to this feature. We're currently using Ghostinspector and miss that feature a lot already.

A failed screenshot is rarely something with a huge severity. Some orderNr, date/time, advertising, rendering differences, animated objects,… has changed. However, just accept 95% matching misses a lot important stuff. Treshold is IMHO not useful at all and should be always 100%.

We are using this method:
- a failed assertion is a failed test, a real error
- a failed screenshot is just a warning
- We do not allow a _single_ pixel difference

Ghostinspector returns two results. Passed and ScreenshotPassed. First on is an error, last one is a warning.

In daily business we have a lot of findings in the screenshots, but like 95% of „looks different“ is a false positive.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you have additional information not present in this issue that you think will help prioritizing it, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.
